tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/TypesTest.java

```diff
@@ -0,0 +1,345 @@
+
+    @BeforeClass
+    public void setup() {
+        CompileResult compileResult = BCompileUtil.compileAndCacheBala("test-src/typesbir");
+        if (compileResult.getErrorCount() != 0) {
+            Arrays.stream(compileResult.getDiagnostics()).forEach(System.out::println);
+            Assert.fail("Compilation contains error");
+        }
+
+        project = BCompileUtil.loadProject("test-src/types-project");
+        model = getDefaultModulesSemanticModel(project);
+        types = model.types();
+    }
+
+    @Test(dataProvider = "BuiltInTypesProvider")
+    public void testBuiltInTypes(TypeSymbol typeSymbol, TypeDescKind kind, Class<? extends TypeSymbol> typeClass) {
+        assertEquals(typeSymbol.typeKind(), kind);
+        assertEquals(typeSymbol.getClass(), typeClass);
+    }
+
+    @DataProvider(name = "BuiltInTypesProvider")
+    private Object[][] getBuiltInTypes() {
+        return new Object[][] {
+                {types.BOOLEAN, BOOLEAN, BallerinaBooleanTypeSymbol.class},
+                {types.INT, INT, BallerinaIntTypeSymbol.class},
+                {types.FLOAT, FLOAT, BallerinaFloatTypeSymbol.class},
+                {types.DECIMAL, DECIMAL, BallerinaDecimalTypeSymbol.class},
+                {types.STRING, STRING, BallerinaStringTypeSymbol.class},
+                {types.NIL, NIL, BallerinaNilTypeSymbol.class},
+                {types.XML, XML, BallerinaXMLTypeSymbol.class},
+                {types.ERROR, ERROR, BallerinaErrorTypeSymbol.class},
+                {types.FUTURE, FUTURE, BallerinaFutureTypeSymbol.class},
+                {types.FUNCTION, FUNCTION, BallerinaFunctionTypeSymbol.class},
+                {types.TYPEDESC, TYPEDESC, BallerinaTypeDescTypeSymbol.class},
+                {types.HANDLE, HANDLE, BallerinaHandleTypeSymbol.class},
+                {types.STREAM, STREAM, BallerinaStreamTypeSymbol.class},
+                {types.ANY, ANY, BallerinaAnyTypeSymbol.class},
+                {types.ANYDATA, ANYDATA, BallerinaAnydataTypeSymbol.class},
+                {types.NEVER, NEVER, BallerinaNeverTypeSymbol.class},
+                {types.READONLY, READONLY, BallerinaReadonlyTypeSymbol.class},
+                {types.JSON, JSON, BallerinaJSONTypeSymbol.class},
+                {types.BYTE, BYTE, BallerinaByteTypeSymbol.class},
+                {types.COMPILATION_ERROR, COMPILATION_ERROR, BallerinaCompilationErrorTypeSymbol.class},
+        };
+    }
+
+    @Test(dataProvider = "TypesByNameProvider")
+    public void testTypeByName(String moduleName, String typeDefName, SymbolKind symKind, TypeDescKind typeDescKind) {
+        Module module = getModule(project, moduleName);
+        model = project.currentPackage().getCompilation().getSemanticModel(module.moduleId());
+        types = model.types();
+        String org = module.descriptor().org().value();
+        String pkgName = module.descriptor().name().toString();
+        String version = module.descriptor().version().toString();
```

Comment:
What's the need for taking these values from the module descriptor? Wouldn't just sending them as params do? Plus then you know what you are testing for. In this case, if these values change for some reason, the test changes too.